pkg/handlers/app.go

```diff
@@ -421,50 +425,97 @@ func (h *Handler) RemoveApp(w http.ResponseWriter, r *http.Request) {
 	JSON(w, http.StatusOK, response)
 }
 
-type IsKotsVersionCompatibleWithAppRequest struct {
-	AppSpec   string `json:"appSpec"`
-	IsInstall bool   `json:"isInstall"`
+type CanInstallAppVersionRequest struct {
+	AppSpec    string `json:"appSpec"`
+	AirgapSpec string `json:"airgapSpec"`
+	IsInstall  bool   `json:"isInstall"`
 }
 
-type IsKotsVersionCompatibleWithAppResponse struct {
-	IsCompatible bool   `json:"isCompatible"`
-	Error        string `json:"error,omitempty"`
+type CanInstallAppVersionResponse struct {
+	CanInstall bool   `json:"canInstall"`
+	Error      string `json:"error,omitempty"`
 }
 
-func (h *Handler) IsKotsVersionCompatibleWithApp(w http.ResponseWriter, r *http.Request) {
-	response := IsKotsVersionCompatibleWithAppResponse{}
+func (h *Handler) CanInstallAppVersion(w http.ResponseWriter, r *http.Request) {
+	appSlug := mux.Vars(r)["appSlug"]
+
+	response := CanInstallAppVersionResponse{
+		CanInstall: false,
+	}
 
-	request := IsKotsVersionCompatibleWithAppRequest{}
+	request := CanInstallAppVersionRequest{}
 	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
 		response.Error = "failed to parse request body"
 		logger.Error(errors.Wrap(err, response.Error))
 		JSON(w, http.StatusBadRequest, response)
 		return
 	}
 
-	if request.AppSpec == "" {
-		response.IsCompatible = true
-		JSON(w, http.StatusOK, response)
-		return
-	}
+	if request.AppSpec != "" {
+		response.CanInstall = false
 
-	kotsApp, err := kotsutil.LoadKotsAppFromContents([]byte(request.AppSpec))
-	if err != nil {
-		response.Error = "failed to load kots app from contents"
-		logger.Error(errors.Wrap(err, response.Error))
-		JSON(w, http.StatusInternalServerError, response)
-		return
-	}
+		kotsApp, err := kotsutil.LoadKotsAppFromContents([]byte(request.AppSpec))
+		if err != nil {
+			response.Error = "failed to load kots app from contents"
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
 
-	if kotsApp != nil {
-		response.IsCompatible = kotsutil.IsKotsVersionCompatibleWithApp(*kotsApp, request.IsInstall)
-	} else {
-		response.IsCompatible = true
+		if kotsApp != nil {
```

Comment:
if kotsApp is nil, `CanInstall` should be explicitly set to `true` (same as old behavior). otherwise it will return an error in the if statement below.